app/contact/page.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import Layout from '@/components/Layout';
+
+const ContactPage: React.FC<any> = () => {
+    const [formState, setFormState] = useState<any>({
+        name: '',
+        email: '',
+        message: ''
+    });
```

Comment:
useState could be replaced with useFormState/useActionState depending on React version if using a server action: https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#forms
Form and its state should be extracted into separate client component while the page component should be a server component.